pkg/autoscaler/stats_scraper.go

```diff
@@ -127,11 +133,54 @@ func (s *ServiceScraper) Scrape() (*StatMessage, error) {
 		return nil, nil
 	}
 
-	stat, err := s.sClient.Scrape(s.url)
-	if err != nil {
-		return nil, err
+	sampleSize := populationMeanSampleSize(readyPodsCount)
+	statCh := make(chan *Stat, sampleSize)
+	defer close(statCh)
+
+	var (
+		avgConcurrency        float64
+		avgProxiedConcurrency float64
+		reqCount              int32
+		proxiedReqCount       int32
+		successCount          float64
+
+		waitGroup sync.WaitGroup
+	)
+
+	waitGroup.Add(sampleSize)
+	for i := 0; i < sampleSize; i++ {
+		go func() {
+			defer waitGroup.Done()
+
+			if stat, err := s.sClient.Scrape(s.url); err != nil {
+				statCh <- nil
+			} else {
+				statCh <- stat
+			}
+		}()
 	}
 
+	waitGroup.Wait()
```

Comment:
opting for the `nil on error` approach here, I think you don't need the WaitGroup at all.

If you want to keep the waitGroup, you can opt to not send nils above, close the channel after the waitgroup is finished and then iterate over the channel like so:

```golang
waitGroup.Wait()
close(statCh)
for _, stat := range statCh {
...
}
```

I have no preference for either way, but please remove the WaitGroup if you want to keep sending nils.

Reply:
Kept the waitGroup and removed sending nils.